app/Jobs/CreateArticle.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Http\Requests\ArticleRequest;
+use App\Models\Article;
+use App\Models\Series;
+use App\User;
+
+final class CreateArticle
+{
+    private $title;
+
+    private $body;
+
+    private $author;
+
+    private $canonicalUrl;
+
+    private $tags;
+
+    private $series;
+
+    public function __construct(string $title, string $body, User $author, string $canonicalUrl = null, array $tags = [], string $series = null)
```

Comment:
I'm not happy with this as there are too many parameters and there will be more to come. Planning a refactor later down the line.

I think it will probably accept an array of attributes instead as we do with `CreateThread`, but I don't like not having control over what is passed in so going to think on it a little more.

Reply:
What I suggest doing is keeping the title, body, author because they're required but add the others as an `$options` array and `$options['tags'] ?? null` as a whitelist.